File: longestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string> &strs) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        if(strs.size()==0)
        {
        	return "";
        }
        int i,j,pos;
        char c;
        for(i=0;i<strs.size();i++)
        {
        	if(strs[i].size()==0)
        	{
        		return "";
        	}
        }

        for(i=0;i<strs[0].size();i++)
        {
        	c = strs[0][i];
        	for(j=1;j<strs.size();j++)
        	{
        		if(strs[j].size()<i+1 || strs[j][i]!=c)
        		{
        			return strs[0].substr(0,i);
        		}
        	}
        }
        return strs[0].substr(0,i);
    }
};

File: longestCommonPrefix.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int len = strs.length;
        if(len == 0) return "";
        if(len == 1) return strs[0];
        String res = strs[0];
        int k = 0;
        while(k < res.length())
        {
        	char c = res.charAt(k);
        	for(int i = 1;i < len;i++)
        	{
        		if(k == strs[i].length())
        			return res.substring(0,k);
        		if(c != strs[i].charAt(k))
        			return res.substring(0,k);
        	}
        	k++;
        }
        return res.substring(0,k);
    }
}